st3/sublime_lib/fancy_settings.py

```diff
@@ -0,0 +1,47 @@
+class FancySettings():
+    def __init__(self, settings):
+        self.settings = settings
+
+    def get(self, name, default=None):
+        return self.settings.get(name, default)
+
+    def set(self, name, value):
+        self.settings.set(name, value)
+
+    def erase(self, name):
+        self.settings.erase(name)
+
+    def has(self, name):
+        return self.settings.has(name)
+
+    def __getitem__(self, key):
+        return self.get(key)
+
+    def __setitem__(self, key, value):
+        self.set(key, value)
+
+    def __delitem__(self, key):
+        self.erase(key)
+
+    def __contains__(self, item):
+        return self.has(item)
+
+    def subscribe(self, key, selector, callback):
+        if isinstance(selector, str):
```

Comment:
I suppose allowing sequences of strings as the selector and building the selector based on `selector = lambda this: {key: this[key] for key in selector}` would also be useful?

Reply:
Implemented.